### PR TITLE
docs: Remove old upgrade info

### PR DESCRIPTION
**Pull request by krantzinator**
This removes all the notes about the differences when upgrading from 0.2.0 and previous versions to 0.2.1 and later versions. This also adds a note about using server snapshots in the `Manual Upgrade` section.

### CODE REVIEW

File: website/content/docs/upgrading/index.mdx [hunk 1]
```diff
@@ -96,9 +75,13 @@ client, and entrypoints in any order for a standard, backwards compatible upgrad
...
 
-2. Shut down the previous server version A.
+2. [Take a snapshot of the server](https://www.waypointproject.io/commands/server-snapshot)
+
+3. Shut down the previous server version A.
+
+4. Start the new server version B.
 
-3. Start the new server version B.
+5. [Restore the server snapshot](https://www.waypointproject.io/commands/server-restore)
```
Comment (catsby) [inline website/content/docs/upgrading/index.mdx:84]: nit: should we add a next step like "verify the connection with `waypoint context verify` or something?

**Review by catsby — APPROVED** [review]
LGTM

**Review by briancain — APPROVED** [review]
Looks good! +1 to what @catsby said, a `waypoint context verify` would be good to add in the manual upgrade section.

File: website/content/docs/upgrading/index.mdx (hunk 1 above)
Comment (krantzinator) [reply website/content/docs/upgrading/index.mdx:84]: Sorry, I smashed the merge button before I saw this 😔 https://github.com/hashicorp/waypoint/pull/2236

**Pull request merged**